src/image/loading_displaying.js

```diff
@@ -1097,6 +1097,7 @@ p5.prototype.image = function(
   img,
   dx,
   dy,
+  dz,
```

Comment:
Thank you for the review. I went with the length approach initially .

Odd number of parameters → Use 2D mode.
Even number of parameters → Check if the 10th parameter is a number.
If it's not a number, defer to 2D mode.
If it's a number, use 3D mode.

but starting from the third parameter  single  arguments become optional `image(img, dx, dy, dWidth, dHeight, sx, sy, [sWidth], [sHeight], [fit], [xAlign], [yAlign])` hence my even/odd length logic fails as soon as the function starts receiving optional arguments.  

![Screenshot from 2024-08-16 19-26-04](https://github.com/user-attachments/assets/e230b182-7a5d-40e0-a0ac-38c2ce639d8a)

The best solution I got at the moment was, to **shift the values of the input argument to the right** in case of 2D_rendering.

```js
  if(this._renderer instanceof p5.RendererGL == false){
    // Store the value of 'yAlign' temporarily
    let   temp = yAlign;
    // From the 3rd arguement shift the assingment one position to the right   
    yAlign = xAlign;
    xAlign = fit;
    fit = sHeight;
    sHeight = sWidth;
    sWidth = sy;
    sy = sx;
    sx = dHeight;
    dHeight = dWidth;
    dWidth = dz;    
  }
```
The results have been successful in every test .

![Screenshot from 2024-08-16 19-31-01](https://github.com/user-attachments/assets/a99bccd8-7e0d-432d-a2b8-d78dcb7bd3b2)

```js
p5.prototype.image = function(
  img,
  dx,
  dy,
  dz,
  dWidth,
  dHeight,
  sx,
  sy,
  sWidth,
  sHeight,
  fit,
  xAlign,
  yAlign
) {
  p5._validateParameters('image', arguments);

  if(this._renderer instanceof p5.RendererGL == false){
    // Store the value of 'yAlign' temporarily
    let   temp = yAlign;
    // From the 3rd arguement shift the assingment one position to the right   
    yAlign = xAlign;
    xAlign = fit;
    fit = sHeight;
    sHeight = sWidth;
    sWidth = sy;
    sy = sx;
    sx = dHeight;
    dHeight = dWidth;
    dWidth = dz;    
  }

  let defW = img.width;
  let defH = img.height;
  yAlign = yAlign || constants.CENTER;
  xAlign = xAlign || constants.CENTER;

  if (img.elt) {
    defW = defW !== undefined ? defW : img.elt.width;
    defH = defH !== undefined ? defH : img.elt.height;
  }
  if (img.elt && img.elt.videoWidth && !img.canvas) {
    // video no canvas
    defW = defW !== undefined ? defW : img.elt.videoWidth;
    defH = defH !== undefined ? defH : img.elt.videoHeight;
  }

  let _dx = dx;
  let _dy = dy;
  let _dw = dWidth || defW;
  let _dh = dHeight || defH;
  let _sx = sx || 0;
  let _sy = sy || 0;
  let _sw = sWidth !== undefined ? sWidth : defW;
  let _sh = sHeight !== undefined ? sHeight : defH;

  _sw = _sAssign(_sw, defW);
  _sh = _sAssign(_sh, defH);

  let vals = canvas.modeAdjust(_dx, _dy, _dw, _dh, this._renderer._imageMode);
  vals = _imageFit(
    fit,
    xAlign,
    yAlign,
    vals.x,
    vals.y,
    vals.w,
    vals.h,
    _sx,
    _sy,
    _sw,
    _sh
  );
  //if it is a WEGL instance default use 3D rendering
  if (this._renderer instanceof p5.RendererGL) {
    this._renderer.image3D(
      img,
      vals.sx,
      vals.sy,
      dz,
      vals.sw,
      vals.sh,
      vals.dx,
      vals.dy,
      vals.dw,
      vals.dh
    );
  }else {
    // tint the image if there is a tint
    // Default 2D rendering
    this._renderer.image(
      img,
      vals.sx,
      vals.sy,
      vals.sw,
      vals.sh,
      vals.dx,
      vals.dy,
      vals.dw,
      vals.dh
    );
  }
};

```
should I go ahead with this approach.

Reply:
Does this also work for 3D mode if the z coordinate is omitted?

Also a possible other approach for the even/odd check: you might need to check if the number of arguments up to the `fit` parameter (which should be a string instead of a number) is even or odd. That might mean something like:
```js
let endIndex = args.findIndex(arg => arg === constants.COVER || arg === constants.CONTAIN)
if (endIndex === -1) {
  endIndex = args.length
}
const argsUpToFit = args.slice(0, endIndex)
```

Reply:
>Does this also work for 3D mode if the z coordinate is omitted?

yes.

Reply:
If you are in WebGL mode and call `image` without `z` but with some of the options after `z`, does that still work? e.g. this should draw the image fullscreen on the canvas:
```js
imageMode(CENTER)
image(yourImg, 0, 0, width, height, 0, 0, yourImg.width, yourImg.height, COVER)
```
Just based on the code so far, it looks like in WebGL mode, it never shifts all the arguments over the way it does when `this._renderer instanceof p5.RendererGL === false`.

Reply:
![image](https://github.com/user-attachments/assets/160a3d25-e3bb-4dd0-9a7c-f3fde5795ff8)

It seems like the `ImgeMode(CENTER)` is a trap on WebGL mode. Is there any way to workaround this ?  we could add a new `imageModeGL` that supports `WebGL` coordinates  ?

```